File: 0917-reverse-only-letters/0917-reverse-only-letters.rb
# @param {String} s
# @return {String}
def reverse_only_letters(s)
  letters = s.chars.select { |c| c =~ /[a-z]/i }
  s.chars.map { |c| c =~ /[a-z]/i ? letters.pop : c }.join
end

File: 0917-reverse-only-letters/0917-reverse-only-letters.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseOnlyLetters = function(s) {
    let letters = s.split('').filter(c => /[a-zA-Z]/.test(c));
    let result = s.split('').map(c => /[a-zA-Z]/.test(c) ? letters.pop() : c).join('');
    return result;
};